### PR TITLE
Feat: Custom matchers

### PR DESCRIPTION
**Pull request by brooksbecton**
### Summary

Begins work on custom jest matchers as part of issue #99 

### Test plan

`yarn test`

### CODE REVIEW

**Comment by Esemesek** [comment]
@thymikee I don't think we want this `package-lock.json` committed.

File: src/__tests__/custom-matchers.test.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import React from 'react';
```
Comment (thymikee) [inline src/__tests__/custom-matchers.test.js]: Please add `// @flow` pragma

File: src/custom-matchers/extendExpect.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import extensions from './index';
```
Comment (thymikee) [inline src/custom-matchers/extendExpect.js]: `// @flow`

File: src/custom-matchers/index.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import { toBeDisabled } from './toBeDisabled';
```
Comment (thymikee) [inline src/custom-matchers/index.js]: `// @flow`

File: src/__tests__/custom-matchers.test.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+import { Button, Text, TextInput, View } from 'react-native';
+import { create } from 'react-test-renderer';
+
+describe('custom-matchers', () => {
+  function render(node) {
```
Comment (thymikee) [inline src/__tests__/custom-matchers.test.js]: can we use `render` from this library?

**Comment by brooksbecton** [comment]
- Removed the package-lock :sweat_smile: 
- Added `// @flow`
- Using `react-native-testing-library`'s render function 

I found Esemesek's branch with the custom-matchers here: https://github.com/callstack/react-native-testing-library/tree/Esemesek/custom-matchers/src/matchers

and I used most of it's .toBeEmpty logic because it is way more thorough

**Comment by brooksbecton** [comment]
Removed package-lock.json. 

Were ya'll wanting to get most of the matchers that are in jest-dom in this PR?

**Comment by thymikee** [comment]
I'd rather like to start small :)

File: src/__tests__/custom-matchers.test.js [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    describe('Button', () => {
+      const BUTTON_EXPLICIT_DISABLED = 'button-explicit-disabled';
+      const BUTTON_IMPLICIT_DISABLED = 'button-implicit-disabled';
+      const BUTTON_EXPLICIT_ENABLED = 'button-explicit-enabled';
+      const BUTTON_IMPLICIT_ENABLED = 'button-implicit-enabled';
+      const TestComponent = () => (
+        <View>
+          <Button
+            testID={BUTTON_EXPLICIT_DISABLED}
+            disabled={true}
```
Comment (thymikee) [inline src/__tests__/custom-matchers.test.js]: This case can be removed, it's the same as below from jsx point of view

File: src/custom-matchers/toBeEmpty.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
+// @flow
+
+function isEmpty(instance: ReactTestInstance) {
+  if (instance.props.children instanceof Array) {
```
Comment (thymikee) [inline src/custom-matchers/toBeEmpty.js]: We should use React.Children for iteration over children

Comment (brooksbecton) [reply src/custom-matchers/toBeEmpty.js]: Are you saying we should check to see if it is instanceof React.Children? 
I don't understand

File: src/__tests__/custom-matchers.test.js (hunk 5 above)
Comment (brooksbecton) [reply src/__tests__/custom-matchers.test.js]: Removed the 'BUTTON_EXPLICIT_DISABLED' test case

**Comment by thymikee** [comment]
Thank you so much for your time! I'll close this in favor of #298 though. We're going to use `@testing-library/jest-native`. I hope it will only be a matter of documenting it and providing some tests that we're compatible :). If we want to add more matchers, we can contribute it back there :)

**Pull request closed without merging**